service_crategen/src/commands/generate/codegen/query.rs

```diff
@@ -240,6 +240,18 @@ fn value_name(service: &Service, shape: &Shape) -> String {
 }
 
 fn member_location(service: &Service, member: &Member, default: &str) -> String {
+    // Seems a bit hacky to avoid doing this just for EC2:
+    match service.shape_for_member(member) {
+        Some(ref shape) => {
+            if service.protocol() != "ec2" && !shape.is_primitive() {
+                if shape.member.is_some() {
```

Comment:
Could this be `if let Some(ref shape_member) = shape.member` to avoid the `unwrap()` below?